manager/integration/tests/test_snapshot.py

```diff
@@ -0,0 +1,590 @@
+                assert str(df['timestamp'][0]) == str(df['timestamp'][1])
+
+            if snapshots.index(s) == len(snapshots)-1:
+                return True
+
+        time.sleep(RETRY_INTERVAL)
+
+    return False
+
+
+def get_checksum_in_checksum_file(disk_path):
+    checksum_file = disk_path + ".checksum"
+    try:
+        f = open(checksum_file)
+        metadata = json.load(f)
+    except OSError as e:
+        print(e)
+        raise
+
+    return metadata["checksum"]
+
+
+def get_ctime_in_checksum_file(disk_path):
+    checksum_file = disk_path + ".checksum"
+    try:
+        f = open(checksum_file)
+        metadata = json.load(f)
+    except OSError as e:
+        print(e)
+        raise
+
+    return metadata["change_time"]
+
+
+def check_snapshot_checksums_set(volume):
+    value = volume.snapshotList()
+    snapshots = value.data
+    for s in snapshots:
+        if s.name == "volume-head":
+            continue
+        if s.checksum == "":
+            return False
+
+    for s in snapshots:
+        print(f'Checksum of snapshot {s.name} is {s.checksum}')
+    return True
+
+
+def get_local_host_replica_data_path(volume):
+    lht_hostId = get_self_host_id()
+
+    dataPath = ""
+    replicas = volume.replicas
+    for r in replicas:
+        if r.hostId == lht_hostId:
+            dataPath = r.dataPath
+            break
+
+    return dataPath
+
+
+def get_checksum_from_snapshot_disk_file(data_path, snapshot_name):
+    exec_cmd = ["java", "-jar", "/jacksum.jar", "-a", "crc64_go-iso",
+                os.path.join(data_path,
+                             "volume-snap-" + snapshot_name + ".img")]
+    try:
+        output = subprocess.check_output(exec_cmd)
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+        return ""
+
+    data = str(output, 'utf-8')
+    fields = data.split(' ')
+    return fields[0]
+
+
+def get_ctime_from_snapshot_disk_file(data_path, snapshot_name):
+    exec_cmd = ["stat", "-c", "%z",
+                os.path.join(data_path,
+                             "volume-snap-" + snapshot_name + ".img")]
+    try:
+        output = subprocess.check_output(exec_cmd)
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+        return ""
+
+    return str(output, 'utf-8').rstrip('\n') + " UTC"
+
+
+def create_snapshots(client, volume, data_size, num_snapshots):  # NOQA
+    dev = get_volume_endpoint(volume)
+
+    snaps = {}
+    for i in range(num_snapshots):
+        assert write_device_random_data(dev, data_size)
+        snaps[i] = create_snapshot(client, volume.name)
+
+    snapshots = volume.snapshotList()
+    snapMap = {}
+    for snap in snapshots:
+        snapMap[snap.name] = snap
+
+    for i in range(num_snapshots):
+        snap = snaps[i]
+        assert snapMap[snap.name].name == snap.name
+        assert snapMap[snap.name].removed is False
```

Comment:
Noticed the assert for `snapMap[snap.name].parent` is removed. Is this intended?

Reply:
Removing it won't influence the test cases. But, I added them back.